joomla/joomla-cms
# [4.0] Add FA svg sprite set(s) & helperclass

Pull Request for Issue # .

### Summary of Changes
adds fontawesome's svg sprite sets to core.
provides a htmlhelper class to handle creating icons from sprites.

### Testing Instructions
apply pr
run npm ci
check that joomla logo in admin hasn't changed looked.
inspect code notice it's now an inline svg, will fill support


### Actual result BEFORE applying this Pull Request
svg images are not available in /media/vendor/fontawesome-free
![image](https://user-images.githubusercontent.com/1850089/98608564-1877eb00-22b1-11eb-8ce1-369c9e6cae1d.png)
![image](https://user-images.githubusercontent.com/1850089/98608553-11e97380-22b1-11eb-8a7a-a78dd1ddce91.png)


### Expected result AFTER applying this Pull Request
![image](https://user-images.githubusercontent.com/1850089/98608414-c040e900-22b0-11eb-995c-a432882ea4ba.png)
![image](https://user-images.githubusercontent.com/1850089/98608437-cfc03200-22b0-11eb-9eb6-ef57c0650eaa.png)


### Documentation Changes Required
svg sprites are used by calling the htmlhelper "sprite" class.
`HTMLHelper::_('sprite', 'icon-joomla', 'brands', '', 'Joomla version', 'Icon showing Joomla\'s current version')`

This class takes 5 arguments.
`string $icon, string $brand = 'solid', string $class = '', string $title = '', string $description = ''`

- icon name. ( can be any name but if icon- is included then it is stripped )
- sprite set name ( brands, regular or solid ) ( 'solid' by default )
- class ( 'icon' + icon name by default )(optional)
- title ( used for tooltip and a11y ) ( optional )
- description ( used by a11y ) (optional)

It returns a string consisting of the html to display the svg as mentioned [here](https://fontawesome.com/how-to-use/on-the-web/advanced/svg-sprites)

default svg icon styling is set in 'administrator/templates/atum/scss/blocks/_icons.scss' by 
```
svg.icon {
  max-height: 2rem;
  max-width: 2rem;
}
```
then individual icons are styled like this..
```
// Joomla
svg.icon-joomla {
  fill: $white;
  width: 1.5rem;
  height: 1.5rem;
}
```
